### ISSUE TITLE
New postUpdateOption - cargoUpdate

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#23971]
### Describe the proposed change(s).

A new postUpdateOption `cargoUpdate` which runs `cargo update` after any Cargo artifacts updating (kind of similar reasoning as `go mod tidy` for `go`).

### Describe why we need/want these change(s).

For those who pin their dependencies in particular, this would mean the lock file is always refreshed to latest transitive dependencies.

We should make sure to note the downsides/reasons you don't want to do this though:
1. This may also result in bumping/upgrading of direct dependencies too (e.g. if constraints are in use in Cargo.toml)
2. This will also result in "unrelated" changes in the lock file. e.g. the PR is for upgrading direct dependency A, but also a lot of transitive dependencies of B are upgraded too, and it changes behavior or breaks the build unrelated to A

**Comment by jhoover4** [comment]
Is there any movement on this?

**Comment by ShaddyDC** [comment]
My use case is somewhat related, but I can also open a new issue for it. We build with the `--locked` flag for reproducibility, and then the CI often fails on renovate updates with this error:
```
> cargo build --all-features --workspace --all-targets --bins  --examples --tests --benches --locked
    Updating crates.io index
error: the lock file /builds/my-user/my-project/Cargo.lock needs to be updated but --locked was passed to prevent this
If you want to try to generate the lock file without accessing the network, remove the --locked flag and use --offline instead.
```

I usually just manually run the build command on the branch without `--locked`, squash the changes, and push so that the CI can run.

I'm fine with not having the transitive dependencies updated each time like mentioned in the op because we use the lockfile maintenance feature, but it would be really nice if we could automate these updates without requiring manual intervention.

Running `cargo update` would address the issue, though it would also update more dependencies than the current workflow, which I'm also not a huge fan of.

**Comment by rarkins** [comment]
@ShaddyDC I was recently looking into `cargo fetch` and wonder if this is a better option

**Comment by ShaddyDC** [comment]
You are correct, running `cargo fetch` seems to be doing what I need! So I guess a `cargoFetch` `postUpdateOption` would be exactly what I'm looking for :)

I also didn't know about `postUpdateOptions` before either, so I guess I can see if that's an option for us.
